go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -0,0 +1,179 @@
+//go:build linux
+// +build linux
+
+package ovspinning
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog/v2"
+)
+
+// These variables are meant to be used in unit tests
+var tickDuration time.Duration = 10 * time.Second
+var getOvsVSwitchdPIDFn func() (string, error) = util.GetOvsVSwitchdPID
+var getOvsDBServerPIDFn func() (string, error) = util.GetOvsDBServerPID
+var featureEnablerFile string = "/etc/openvswitch/enable_dynamic_cpu_affinity"
+
+// Run monitors OVS daemon's processes (ovs-vswitchd and ovsdb-server) and sets their CPU affinity
+// masks to that of the current process.
+// This feature is enabled by the presence of a non-empty file in the path `/etc/openvswitch/enable_dynamic_cpu_affinity`
+func Run(stopCh <-chan struct{}) {
+
+	klog.Infof("Starting OVS daemon CPU pinning")
+
+	fileWatcher, err := createFileWatcherFor(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
+		return
+	}
+	ticker := time.NewTicker(tickDuration)
+
+	// If the file exists and is not empty, enable the affinity pinning
+	isFeatureEnabled, err := isFileNotEmpty(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Error while reading [%s]: %v", featureEnablerFile, err)
+		isFeatureEnabled = false
+	}
+
+	for {
+		select {
+		case _, ok := <-fileWatcher.Events:
```

Comment:
you may want to differentiate between events here. For example, if the file is removed with a remove event, the watcher will no longer track the file. If you care about continuing to monitor the file you need to add it back:
https://github.com/ovn-org/ovn-kubernetes/pull/2888/files#diff-df866e3dd08aef582047e1434567f0a75569dcedb9d7ece04437a1746dbc3effR196

Otherwise if someone removes the file, maybe you just stop monitoring all together and print a  log saying restart of ovnkube-node is required to renable pinning or something.

Reply:
Thanks, I wasn't aware of that fsnotify behavior. 

> Otherwise if someone removes the file, maybe you just stop monitoring all together and print a log saying restart of ovnkube-node is required to renable pinning or something.

I'll go for this, it looks like the simplest solution.